File: 1040/1040 - Média 3.lua
a,b,c,d = io.read("*n"),io.read("*n"),io.read("*n"),io.read("*n")

mp = (a*2+b*3+c*4+d*1)/10
print('Media: '..string.format("%.1f", mp))
if mp >= 7.0 then print('Aluno aprovado.') end
if mp < 5 then print('Aluno reprovado.') end
if mp >= 5.0 and mp <= 6.9 then 
    print('Aluno em exame.')
    n = io.read("*n")
    print('Nota do exame: '..string.format("%.1f", n))
    m = (mp+n)/2
    if m >= 5.0 then print('Aluno aprovado.') end
    if m <= 4.9 then print('Aluno reprovado.') end
    print('Media final: '..string.format("%.1f", m)) end

File: 1040/1040 - Média 3.ex
[a, b, c, d] = IO.read(:stdio,:line) |> String.trim |> String.split |> Enum.map(fn str ->
  case Float.parse str  do
    {number, _} -> number
  end
end)

media1 = (a*2 + b*3 + c*4 + d) / Enum.sum [2, 3, 4, 1]
IO.puts "Media: #{:erlang.float_to_binary(media1, [decimals: 1])}"

if media1 >= 7.0, do: IO.puts "Aluno aprovado."
if media1 < 5.0, do: IO.puts "Aluno reprovado."
if media1 >= 5.0 && media1 <= 6.9 do
  IO.puts("Aluno em exame.")
  {n, _} = IO.read(:stdio,:line) |> String.trim |> Float.parse
  IO.puts "Nota do exame: #{:erlang.float_to_binary(n, [decimals: 1])}"
  
  media2 = (media1+n)/2
  if media2 >= 5.0, do: IO.puts "Aluno aprovado."
  if media2 <= 4.9, do: IO.puts "Aluno reprovado."
    IO.puts "Media final: #{:erlang.float_to_binary(media2, [decimals: 1])}"
end
